src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -886,6 +892,47 @@ export class ResultsViewPageStore {
         }
     });
 
+    readonly nonQueriedMolecularData = remoteData<NumericGeneMolecularData[]>({
+        await: () => [
+            this.studyToDataQueryFilter,
+            this.genes,
+            this.nonSelectedMolecularProfiles,
+            this.samples
+        ],
+        invoke: () => {
+            // we get mutations with mutations endpoint, all other alterations with this one, so filter out mutation genetic profile
+            const profilesWithoutMutationProfile = _.filter(this.nonSelectedMolecularProfiles.result, (profile: MolecularProfile) => profile.molecularAlterationType !== 'MUTATION_EXTENDED');
+            const genes = this.genes.result;
+
+            if (profilesWithoutMutationProfile.length && genes != undefined && genes.length) {
+
+                const identifiers : SampleMolecularIdentifier[] = [];
+
+                profilesWithoutMutationProfile.forEach((profile:MolecularProfile)=>{
```

Comment:
you can do this operation more efficiently by putting all studyIds in a map and then samples.result.filter( (sample)=>sample.studyId in myMap).

Reply:
remember try to avoid declaring an array and then using forEach to populate it.  usually it's more clear and efficient to use filter or reduce or map.

Reply:
Thanks for the advice!